### ISSUE TITLE
System.ObjectDisposedException when trying to fetch users in 7.0

### ISSUE THREAD

**Issue by mattgrande** [auth0/auth0.net#366]
### Description

I am currently unable to fetch users via the Management API; I get a `System.ObjectDisposedException`, even though the object should not have been disposed.

This is a new application that is using v7.0; Downgrading to 6.5.6 solves the issue. Therefore, I believe this is a regression.

Please note that I originally assume this was my issue, and created a thread on community.auth0.com here: https://community.auth0.com/t/unable-to-connect-to-net-management-api/37462

### Reproduction

    public async void GetUsers()
    {
        using (var managementApi = GetManagementApi())
        {
            var users = await managementApi.Users.GetUsersByEmailAsync("a-valid-email@example.com");
            Console.WriteLine(users);
        }
    }

    private ManagementApiClient GetManagementApi()
    {
        var authToken = GetManagementAuthToken();
        return new ManagementApiClient(authToken, new Uri($"https://{Domain}/api/v2/"));
    }

I can provide the code for `GetManagementAuthToken` if necessary, however it's successfully returning a valid token, so I don't believe there to be an issue there.

The line with the `await` throws the following exception:

    System.ObjectDisposedException: 'Cannot access a disposed object. Object name: 'System.Net.Http.HttpRequestMessage'.'

### Environment

- Library version 7.0.0.0
- .NET Framework 4.7.2
- Windows 10

**Comment by damieng** [comment]
I just tried the following fragment and it worked fine

```csharp
using (var managementApi = new ManagementApiClient(token, GetVariable("AUTH0_MANAGEMENT_API_URL")))
{
    var users = await managementApi.Users.GetUsersByEmailAsync("damieng+test1@gmail.com");
    Console.WriteLine(users);
}
```

At a guess I would suspect that your GetManagementApi function is returning a shared instance of ManagementApiClient?

**Comment by damieng** [comment]
Also tried your two functions inside a unit test and it also passed.  Confused how this is going wrong.

**Comment by mattgrande** [comment]
Hm, very strange. Is your `managementApiUrl` a string, or Uri object? Looking at the code, I don't _think_ that would make a difference, but I'm just spitballing...

**Comment by damieng** [comment]
It was a string, I just wrapped it in a `new Uri` and it didn't fail.

What kind of appliction are you developing?  It this UWP, WPF, ASP.NET?...

**Comment by mattgrande** [comment]
ASP.Net

**Comment by damieng** [comment]
To get unblocked for now until we figure this out I'd recommend following the guidance at https://auth0.github.io/auth0.net/#client-reuse-and-thread-safety which lets you register a single HttpClientManagementConnection that is shared between ManagementApiClients.  That way you no longer need to use Dispose/`using` statements at all and actually get better performance as the HttpClient is shared/pooled between instances.

**Comment by mattgrande** [comment]
Okay, for the time being I've downgraded to the old version, but at some point in the next day or two, I'll try doing it that way.

Thanks for your help thus far! 👍

**Comment by mark-dispatch** [comment]
For what it's worth, I'm getting the same error calling `Clients.GetAsync()` with the v7.0 library and .NET Framework 4.7.2.

**Comment by humayun-ahmed** [comment]
Yes v7.0 works for .Net core application but for .Net framework, it doesn't work. 

If you look over the code, SendRequest is a task and after that using statements ends which actually disposing the request. After dispose of request, task SendRequest executes where request object is needed for "var response = await httpClient.SendAsync(request).ConfigureAwait(false);"

using (var request = new HttpRequestMessage(HttpMethod.Get, uri))
            {
                ApplyHeaders(request.Headers, headers);
                return SendRequest<T>(request, converters);
            }

**Comment by damieng** [comment]
Additional fix on #368

**Comment by damieng** [comment]
Fix has been released in 7.0.1 now available on NuGet.

**Comment by mattgrande** [comment]
Thank you, everyone!
